UndertaleModLib/Models/UndertaleRoom.cs

```diff
@@ -511,11 +511,16 @@ public void SetupRoom(bool calculateGridWidth = true, bool calculateGridHeight =
             {
                 if (layer.LayerType == LayerType.Assets)
                     tileList = tileList.Concat(layer.AssetsData.LegacyTiles);
-                else if (layer.LayerType == LayerType.Tiles && layer.TilesData.TileData.Length != 0)
+                else if (layer.LayerType == LayerType.Tiles && layer.TilesData.TileData.Length != 0 && layer.TilesData.Background is not null)
                 {
-                    int w = (int) (Width / layer.TilesData.TilesX);
-                    int h = (int) (Height / layer.TilesData.TilesY);
-                    tileSizes[new(w, h)] = layer.TilesData.TilesX * layer.TilesData.TilesY;
+                    int w = (int)layer.TilesData.Background.GMS2TileWidth;
+                    int h = (int)layer.TilesData.Background.GMS2TileHeight;
+                    Point scale = new(w, h);
+                    uint numTiles = layer.TilesData.TilesX * layer.TilesData.TilesY;
+                    if (tileSizes.ContainsKey(scale))
+                        tileSizes[scale] += numTiles;
+                    else
+                        tileSizes[scale] = numTiles;
```

Comment:
This is correct, but for the future - if the dictionary value is a reference type (e.g. `List<>`), then you should use this for a better performance:
```csharp
if (dict.TryGetValue(key, out var list))
    list.Add(item);
else
    list = new() { item }; // Initialize
```

Reply:
Oops, fixed.

Reply:
Oh, I found an even better way (that works similar, but for the value types) - https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.collectionsmarshal.getvaluereforadddefault?view=net-9.0
But it's only .NET 9+ (UTMT is .NET 8).